clients/spark/core/src/main/hadoop3/scala/io/treeverse/clients/conditional/GCBackupAndRestore.scala

```diff
@@ -81,13 +86,53 @@ object GCBackupAndRestore {
   // Construct a DistCp command of the form:
   // `hadoop distcp -D<per storage credentials> -f absoluteAddressesTextFilePath dstNamespaceForHadoopFs`
   // example for command format https://docs.lakefs.io/integrations/distcp.html
+  // with distCp options from https://hadoop.apache.org/docs/r3.2.1/hadoop-distcp/DistCp.html#Command_Line_Options
   def constructDistCpCommand(
       hadoopProps: Array[(String, String)],
       absoluteAddressesTextFilePath: String,
-      dstNamespaceForHadoopFs: String
+      dstNamespaceForHadoopFs: String,
+      storageType: String,
+      hc: Configuration,
+      numObjectsToCopy: Long
   ): Array[String] = {
-    hadoopProps.map((prop) => "-D" + prop._1 + "=" + prop._2) ++
+
+    val distCpLogsPath = hc.get(DistCpConstants.CONF_LABEL_LOG_PATH,
+                                s"${dstNamespaceForHadoopFs.stripSuffix("/")}/_distCp/logs/"
+                               )
+    val distCpLogsPathForFS =
+      ApiClient
+        .translateURI(URI.create(distCpLogsPath), storageType)
+        .normalize()
+        .toString
+    // Tune distCp options that control the speed of the file-to-copy list building stage
+    val numListstatusThreads =
+      hc.get(DistCpConstants.CONF_LABEL_LISTSTATUS_THREADS, DistCpMaxNumListStatusThreads.toString)
+    // Tune distCp options that control the speed of the copy stage
+    // https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.1.4/bk_cloud-data-access/content/distcp-perf-mappers.html
+    val maxMaps = hc.get(DistCpConstants.CONF_LABEL_MAX_MAPS,
+                         max(DistCpConstants.DEFAULT_MAPS, numObjectsToCopy / 1000).toString
```

Comment:
Please round up the division.  While we are protected from the worst cases (getting 0...) by DEFAULT_MAPS, if we round up then we know we get up to 1000 objects per map.  Right now we're rounding down, so we know we get at least 1000 objects per map but it's hard to figure out exactly how many.

Reply:
thx, I decided to use the default num of max maps given by distCp so this line changed